src/DistributedCache.AzureTableStorage.V3/Implementations/AzureTableStorageCache.cs

```diff
@@ -16,27 +16,26 @@ namespace DistributedCache.AzureTableStorage.Implementations;
 /// <summary>
 /// An <see cref="IDistributedCache"/> implementation to cache data in Azure Table Storage.
 /// </summary>
-/// <seealso cref="IDistributedCache"/>.
-public class AzureTableStorageCache : IDistributedCache
+internal abstract class AzureTableStorageCache : IDistributedCache
 {
     private static readonly TimeSpan MinimumExpiredItemsDeletionInterval = TimeSpan.FromMinutes(5);
     private readonly ISystemClock _systemClock;
     private readonly TimeSpan? _expiredItemsDeletionInterval;
     private DateTimeOffset _lastExpirationScan;
     private readonly Func<Task> _deleteExpiredCachedItemsDelegate;
     private readonly string _partitionKey;
-    private readonly Lazy<TableClient> _tableClient;
+    private readonly TableClient _tableClient;
 
     /// <summary>
     /// Initializes a new instance of the <see cref="AzureTableStorageCache"/> class.
     /// </summary>
     /// <param name="options">The options.</param>
-    public AzureTableStorageCache(IOptions<AzureTableStorageCacheOptions> options)
+    /// <param name="tableServiceClient">The TableServiceClient.</param>
+    public AzureTableStorageCache(IOptions<AzureTableStorageCacheOptions> options, TableServiceClient tableServiceClient)
     {
-        Guard.NotNull(options, nameof(options));
-
-        var cacheOptions = options.Value;
+        Guard.NotNull(tableServiceClient);
 
+        var cacheOptions = Guard.NotNull(options.Value);
         Guard.NotNullOrEmpty(cacheOptions.ConnectionString);
```

Comment:
Guard requires a ConnectionString when using StorageAccountUri. Though it's easily resolved with `ConnectionString = "..."`, conditional guards might be more elegant

Reply:
Good catch.

Actually this check should be removed here.

Can you review the PR again?

Reply:
@anth12 
Can you review the most recent code?

And if this one is fine, I can create a preview-02 to double check again?